File: Cpp/Minimum_Time_to_Make_Rope_Colorful.cpp
class Solution {
public:
    int minCost(string colors, vector<int>& needed) {
        long long totalTime = 0;
        int maxTime = needed[0];
        for (int i = 1; i < colors.size(); i++) {
            if (colors[i] == colors[i - 1]) {
                totalTime += min(maxTime, needed[i]);
                maxTime = max(maxTime, needed[i]);
            } else {
                maxTime = needed[i];
            }
        }
        return totalTime;
    }
};

File: Rust/Minimum_Time_to_Make_Rope_Colorful.rs
impl Solution {
    pub fn min_cost(colors: String, needed_time: Vec<i32>) -> i32 {
        let mut total_time = 0;
        let mut max_time = needed_time[0];
        let colors: Vec<char> = colors.chars().collect();
        for i in 1..colors.len() {
            if colors[i] == colors[i - 1] {
                total_time += max_time.min(needed_time[i]);
                max_time = max_time.max(needed_time[i]);
            } else {
                max_time = needed_time[i];
            }
        }
        total_time
    }
}
